include/seqan3/search/algorithm/detail/search.hpp

```diff
@@ -48,7 +48,7 @@ namespace seqan3::detail
 
 //!\brief \todo Docs missing
 template <typename index_t, typename query_t, typename config_t>
-inline auto _search_single(index_t const & index, query_t const & query, config_t const & cfg)
+inline auto _search_single(index_t const & index, query_t & query, config_t const & cfg)
```

Comment:
why do these function names begin with an underscore? This violates SeqAn3 style guide.